tools/fetch_remote_content/src/main.rs

```diff
@@ -1,16 +1,45 @@
-use std::env;
-use std::io;
+use std::{
+    env, io, str,
+    sync::{Arc, Mutex},
+};
 extern crate readability;
+use poptea::{GeminiClient, NoTrustStore};
 use readability::extractor;
 
+fn get_http(url: &str) -> Result<String, Box<dyn std::error::Error>> {
+    Ok(extractor::scrape(url)?.text)
+}
+
+fn get_gemini(url: &str) -> Result<String, Box<dyn std::error::Error>> {
+    let gemini_response = poptea::TlsClient::new(Arc::new(Mutex::new(NoTrustStore::default())))
+        .get(&url)
+        .map_err(|e| fetch_remote_err(&e.to_string()))?;
+    Ok(gemini_response
+        .body
+        .map(String::from_utf8)
+        .ok_or("no content")?
+        .map_err(|e| fetch_remote_err(&e.to_string()))?)
+}
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let url = env::args().nth(1).ok_or_else(|| {
-        Box::new(io::Error::new(
-            io::ErrorKind::Other,
-            "a single argument is required",
-        ))
-    })?;
-    let product = extractor::scrape(&url)?;
-    println!("{}", product.text);
+    let url = env::args()
+        .nth(1)
+        .ok_or_else(|| fetch_remote_err("a single argument is required"))?;
+
+    let archive_text = match url
+        .split_once("://")
+        .ok_or_else(|| fetch_remote_err("scheme not provided"))?
+    {
+        ("https" | "http", _) => get_http(&url)?,
+        ("gemini", _) => get_gemini(&url)?,
+        _ => "unsuported protocol".into(),
```

Comment:
better return an error rather than printing "unsupported protocol" to stdout here

Reply:
Fixed and tested.